### PR TITLE
Java: more manual models

### PR DESCRIPTION
**Pull request by owen-mc**
All these commits relate to manual models to fix incorrect data-flow-generated models in https://github.com/github/codeql/pull/14919. The first two commits are fixing bad df-gen models which were found because they caused lots of false positive alerts. The third commit is adding manual models to classes where we already have a lot of manual models, but some methods had df-gen models which didn't fit in well with the manual models. The fourth commit fixes an incorrect type in a model that was added recently. The fifth commit adds models for methods that are very similar to methods we have recently added manual models for in relation to that PR.

### CODE REVIEW

**Review by atorralba — DISMISSED** [review]

**Comment by atorralba** [comment]
There are a few new models. Does it make sense to get this through DCA just in case?

File: java/ql/lib/ext/java.util.model.yml [hunk 1]
```diff
@@ -309,6 +311,9 @@ extensions:
       - ["java.util", "Queue", True, "poll", "()", "", "Argument[this].Element", "ReturnValue", "value", "manual"]
       - ["java.util", "Queue", True, "remove", "()", "", "Argument[this].Element", "ReturnValue", "value", "manual"]
       - ["java.util", "ResourceBundle", True, "getString", "(String)", "", "Argument[this].MapValue", "ReturnValue", "value", "manual"]
+      - ["java.util", "Scanner", True, "findAll", "(Pattern)", "", "Argument[this]", "ReturnValue.Element", "taint", "df-manual"]
+      - ["java.util", "Scanner", True, "findAll", "(String)", "", "Argument[this]", "ReturnValue", "taint", "df-manual"]
```
Comment (aschackmull) [inline java/ql/lib/ext/java.util.model.yml:315]:
```suggestion
      - ["java.util", "Scanner", True, "findAll", "(String)", "", "Argument[this]", "ReturnValue.Element", "taint", "df-manual"]
```

Comment (aschackmull) [reply java/ql/lib/ext/java.util.model.yml:315]: It looked weird that the two `findAll` methods had different return paths.

**Comment by owen-mc** [comment]
DCA showed nothing interesting.

**Comment by owen-mc** [comment]
@aschackmull or @atorralba Please re-review when you have time.

File: java/ql/lib/ext/java.nio.file.model.yml [hunk 2]
```diff
@@ -91,7 +91,7 @@ extensions:
     data:
       # summary neutrals
       - ["java.nio.file", "Files", "exists", "(Path,LinkOption[])", "summary", "manual"]
-      - ["java.nio.file", "Files", "newInputStream", "(Path,LinkOption[])", "summary", "df-manual"]
+      - ["java.nio.file", "Files", "newInputStream", "(Path,OpenOption[])", "summary", "df-manual"]
```
[the remark this answers is not in the document]
Comment (owen-mc) [reply java/ql/lib/ext/java.nio.file.model.yml:94]: It's easily done. I'm slightly surprised there isn't any validation that could catch it.

**Review by atorralba — APPROVED** [review]

File: java/ql/lib/ext/java.lang.model.yml [hunk 3]
```diff
@@ -114,6 +114,7 @@ extensions:
       - ["java.lang", "String", False, "indent", "", "", "Argument[this]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "String", False, "intern", "", "", "Argument[this]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "String", False, "join", "", "", "Argument[0..1]", "ReturnValue", "taint", "manual"]
+      - ["java.lang", "String", False, "lines", "()", "", "Argument[this]", "ReturnValue.Element", "taint", "df-generated"]
```
Comment (atorralba) [inline java/ql/lib/ext/java.lang.model.yml:117]: Should this be `df-manual` instead, since we've reviewed it?

Comment (owen-mc) [reply java/ql/lib/ext/java.lang.model.yml:117]: Doh, yes. I see another 5 together as well.

**Review by atorralba — APPROVED** [review]

**Pull request merged**